Assets/GlitchEscape/Scripts/Interaction/InteractableTank.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+[RequireComponent(typeof(InteractionTrigger))]
+public class InteractableTank : MonoBehaviour, IPlayerInteractable
+{
+    public GameObject playerInTank;
+    public Transform floatTextArea;
+    public FloatingText floatingText;
```

Comment:
make this member private, and get your floatingText instance at startup (Start() or Awake()) via its singleton instance (see below).

This is so that this dependency (which effectively is / should be a singleton) will always be acquired at startup. A prefab *cannot* get this automatically as it'll need a different reference to this object per scene, and if it isn't assigned it'll be null. Prefabs should always be setup so that you can just drag + drop them into a scene and they'll work; as is, this is not the case, but you can fix this by getting this reference at startup (and you can speed this up a bit / make this more efficient by just making this a singleton)

And also yes, Player could / maybe should be a singleton, but oh well. (could be fixed later)